### DISCUSSION TITLE
How to reuse multiple composites?

### DISCUSSION THREAD

**Discussion by blasdfaa** [nuxt/nuxt#21336] [Questions]
I need to show the quantities of products in different components. If I do this in the `.vue` component, everything works fine:
```ts
// useWishlistQty.ts
const useWishlistQty = () => useLazyFetch(...);

// useCart.ts
const useCart = () => useLazyFetch(...);
```

```ts
// user.vue

const { data: wishlist } = await useWishlistQty();
const { data: cart } = await useCart();

const wishlistQty = wishlist.value?.data.qty;
const cartQty = cart.value ? Object.keys(cart.value.data.cart_data).length : 0;
```

But if I define it in another .ts file to reuse, everything breaks.
```ts
// useProductsCount.ts

export async function useProductsCount() {
  const { data: wishlist } = await useWishlistQty();
  const { data: cart } = await useCart();

  const wishlistQty = wishlist.value?.data.qty;
  const cartQty = cart.value ? Object.keys(cart.value.data.cart_data).length : 0;

  return { wishlistQty, cartQty };
}
```

This is how I use
```ts
// user.vue

const { cartQty, wishlistQty } = await useProductsCount();
```

I get an error
`[nuxt] A composable that requires access to the Nuxt instance was called outside of a plugin, Nuxt hook, Nuxt middleware, or Vue setup function. This is probably not a Nuxt bug. Find out more at https://nuxt.com/docs/guide/concepts/auto-imports#using-vue-and-nuxt-composables.`

Maybe I missed something and am using it wrong? Need help :)

**Comment by warflash** [comment]
This seems correct at first glance, if you still have that issue could you put up a reproduction on stackblitz then it's easier to help with things like that

**Reply by blasdfaa** [reply]
> This seems correct at first glance, if you still have that issue could you put up a reproduction on stackblitz then it's easier to help with things like that

https://stackblitz.com/edit/nuxt-starter-xdrh2z?file=app.vue
